dnf/base.py

```diff
@@ -1690,7 +1690,13 @@ def trans_install(query, remove_query, comps_pkg, strict):
                     sltr.set(provides="({} if {})".format(comps_pkg.name, comps_pkg.requires))
                 else:
                     if self.conf.obsoletes:
-                        query = query.union(self.sack.query().filterm(obsoletes=query))
+                        # if there are no installed packages in the query, add only
+                        # obsoleters of the latest versions.
```

Comment:
May I ask you to add information why we should add all obsoletes when installed package is in the selection. It might be worth to document it.